cpp/core_pyos.h

```diff
@@ -32,20 +37,61 @@ class ReadError {
 };
 
 inline Str* GetUserName(int uid) {
-  assert(0);
+  Str *result = kEmptyString;
+
+  if (passwd *pw = getpwuid(uid))
+  {
+    result = new Str(pw->pw_name);
+  }
+
+  return result;
 }
 
 inline Str* OsType() {
-  // uname()[0].lower()
-  return new Str("TODO");
+  Str *result = kEmptyString; // TODO(Jesse): Should we set this to "unknown" or something?
+
+  utsname un = {};
+  if (::uname(&un) == 0)
+  {
+    result = new Str(un.sysname);
+  }
+
+  return result;
 }
 
 inline Tuple3<double, double, double> Time() {
-  assert(0);
+  rusage ru; // NOTE(Jesse): Doesn't have to be cleared to 0.  The kernel clears unused fields.
+  ::getrusage(RUSAGE_SELF, &ru);
+
+  time_t t = time_::time();
+  auto result = Tuple3<double, double, double>((double)t, (double)ru.ru_utime.tv_sec, (double)ru.ru_stime.tv_sec);
+  return result;
 }
 
 inline void PrintTimes() {
-  assert(0);
+  tms t;
+  if (times(&t) == -1)
+  {
+    printf("Error during time()");
```

Comment:
ditto I think we need `errno` 

Actually now I see in `fcntl_.cc` it does `throw new IOError(errno)`, which I think does the right thing.

That exception should be caught an an error printed by the translated Python code